### DISCUSSION TITLE
brew installed octave contains no graphics toolkit, preventing github action headless plots

### DISCUSSION THREAD

**Discussion by fangq** [Homebrew/discussions#5687] [Linux] [answered]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

The issue has been reported in Homebrew/brew#18595, github action testing an octave toolbox failed on both `macos-13` and `macos-14` runners

https://github.com/fangq/iso2mesh/actions/runs/11413968520/job/31762253381
https://github.com/fangq/iso2mesh/actions/runs/11408582821/job/31747587033

the error is 
```
error: no graphics toolkits are available!
```

in comparison, other Linux/windows runners all present a default graphics_toolkit,allow CI script containing plots in headless mode. However, it appears that the macos-13 and -14 versions of octave were built differently in brew.

as suggested by @Bo98 in Homebrew/brew#18595, because `macos-12` runner `brew install octave`, especially `brew install qt` took forever to complete due to the stop of support, I would like to migrate my github action to macos-13 and -14, but need your help in resolving the graphics toolkit error above.

is there a way to enable graphics toolkit in brew compiled octave in order to use in github action tests?

**Comment by cho-m** [comment, marked as answer]
I think it should be able to use `fltk` but doesn't seem to be found, e.g.
```console
❯ octave-cli --eval 'available_graphics_toolkits()'
ans = {}(1x0)
```

Most functional option is just standard Qt `octave` which may still work in CI given we do sometimes run tests on GitHub runner.
```console
❯ octave --eval 'available_graphics_toolkits()'
ans =
{
  [1,1] = qt
}
```

It looks like your Ubuntu test runs via `gnuplot`.

On macOS, Octave can use `gnuplot` though output won't actually work unless unmaintained AquaTerm.app (`aquaterm` cask) is installed so not ideal. You can see if installing just `gnuplot` gets you past the error

```console
❯ brew install gnuplot
❯ octave-cli --eval 'available_graphics_toolkits()'
ans =
{
  [1,1] = gnuplot
}
```

**Comment by fangq** [comment]
@cho-m, thank you so much! `brew install gnuplot` solved the problem.

I am wondering why `gnuplot` is not setup as a dependency for `octave`?  without a fall-back graphics toolkit, `octave-cli` won't be able to run any script with plotting functions, which is quite limiting.

**Comment by gromgit** [comment]
> I am wondering why gnuplot is not setup as a dependency for octave?

Perhaps because no one felt the need to use it with `octave`, or because they already had `gnuplot` installed for some other reason, so everything Just Worked?

If you feel `gnuplot` should be a required dependency, you can file a PR: https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request
